components/net/bluetooth_thread.rs

```diff
@@ -17,14 +18,19 @@ use rand::{self, Rng};
 use std::borrow::ToOwned;
 use std::collections::{HashMap, HashSet};
 use std::string::String;
+use std::sync::atomic::{AtomicBool, ATOMIC_BOOL_INIT, Ordering};
 use std::thread;
 use std::time::Duration;
 #[cfg(target_os = "linux")]
 use tinyfiledialogs;
 use util::thread::spawn_named;
 
+static TESTING: AtomicBool = ATOMIC_BOOL_INIT;
```

Comment:
Is there any way you could avoid test-related code in non-test code? (this might mean some interface redesign)